library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java

```diff
@@ -86,8 +98,16 @@ int getBatchStatus(long batchId) {
             }
         }
 
+        boolean hasCompleteItems = statusCountMap.hasCountFor(DownloadStatus.SUCCESS);
+        boolean hasPendingItems = statusCountMap.hasCountFor(DownloadStatus.PENDING);
+        boolean hasSubmittedItems = statusCountMap.hasCountFor(DownloadStatus.SUBMITTED);
+        boolean hasOtherItems = statusCountMap.hasNoItemsWithStatuses(STATUSES_EXCEPT_PENDING_SUCCESS_SUBMITTED);
+        if (hasCompleteItems && (hasPendingItems || hasSubmittedItems) && !hasOtherItems) {
```

Comment:
It makes a bit more sense but I don't fully understand 100%, anyway I trust you and the tests so :+1: from me